### ISSUE TITLE
Improve S1751 to cover case where all branches jump out of the loop

### ISSUE THREAD

**Issue by saberduck** [SonarSource/SonarTS#538]
Currently we don't detect cases like these

```javascript
function foo() {
  for (let item of arr) {
    if (item) {
      return item;
    } else {
      return false;
    }
  }
}
```

Following loop will loop only once, while this is not what was probably intended.
